### PR TITLE
Update DAPLink documentation

### PR DESCRIPTION
**Pull request by linlingao**

### CODE REVIEW

File: docs/AUTOMATED_TESTS.md [hunk 1]
```diff
@@ -1,12 +1,71 @@
...
+ 
+### Running Tests
+ * Open CMD and cd to DAPLink directory.
+ * Activate virtual env, launch tests by running ``python test/run_test.py`
+   Specifying --firmwaredir is optional. For target images, either provide mbed.org authentication or use --targetdir to specify the location of pre-built target app binaries.
+ * Test results will be printed to console
+
+## Test on Linux
+###  Prepare Host Machine
+ Linux has a deamon called modemmanager that sends unsolicitated traffic to DAPLink. This will cause serial port tests to fail, typically at band rate initialization. The easiest way is to remove this deamon with ``sudo apt-get purge modemmanager``
```
Comment (cmonr) [inline docs/AUTOMATED_TESTS.md]: "Linux has a daemon..."

You might want to replace this with one of the following:
"Linux may have a daemon..."
"Ubuntu has a daemon..."

I know of at least one Linux distro that doesn't have this installed by default.

File: docs/AUTOMATED_TESTS.md [hunk 2]
```diff
@@ -1,12 +1,71 @@
...
+
+**Step 1.** Build Firmware Releases
+ The firmware images under test need to be built on Windows. Under tools directory, run ``build_release_uvision.bat``. Then copy the generated  uvision_release directory to Linux.
+ 
+**Step 2.** Install Tools
+ Install [Python 2, 2.7.9 or above](https://www.python.org/downloads/).
```
Comment (cmonr) [inline docs/AUTOMATED_TESTS.md]: Couldn't "Python 2, 2.7.9 or above" be replaced with "Python 2.7.9 or above"?

File: docs/AUTOMATED_TESTS.md (hunk 1 above)
Comment (linlingao) [reply docs/AUTOMATED_TESTS.md]: Good point. Will update the doc.

File: docs/AUTOMATED_TESTS.md [hunk 3]
```diff
@@ -1,12 +1,71 @@
...
+### Prepare Host Machine
+Windows drive indexing needs to be turned off prior to running the tests.
+ 
+* Disable indexing on removable drives 
+ Run gpedit.msc (<Windows>+<R> to open run window) then go to:
+ Computer Configuration \ Administrative Templates \ Windows Components \ Search \ Do not allow locations on removable drives to be added to  libraries. Enable the setting there.
+ 
+### Running Tests
+ * Open CMD and cd to DAPLink directory.
+ * Activate virtual env, launch tests by running ``python test/run_test.py`
```
[the remark this answers is not in the document]
Comment (linlingao) [reply docs/AUTOMATED_TESTS.md]: Good eyes. Will add the missing ` in the next commit.

File: docs/AUTOMATED_TESTS.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (linlingao) [reply docs/AUTOMATED_TESTS.md]: Directions to disable it will be in the next commit.

File: docs/AUTOMATED_TESTS.md (hunk 2 above)
Comment (linlingao) [reply docs/AUTOMATED_TESTS.md]: Actually I think we do need to leave the 2 here, since Python 3 is not supported.

**Pull request merged**